src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -94,10 +94,23 @@ public static AppiumDriverLocalService buildService(AppiumServiceBuilder builder
     }
 
     public AppiumDriverLocalService withBasePath(String basePath) {
-        this.basePath = basePath;
+        this.basePath = sanitizeBasePath(basePath);
         return this;
     }
 
+    @SneakyThrows private static String sanitizeBasePath(String basePath) {
+        if (null == basePath) {
+            LOG.warn("Base Path cannot be NULL -- ignoring the basepath configuration");
+            return null;
+        }
+        basePath = basePath.trim();
+        if (basePath.isBlank() || basePath.isEmpty()) {
```

Comment:
`String#isBlank` is available starting from Java 11 only.

Reply:
Apache commons lib also provides isBlank method

Reply:
quick question on this - do we have any log/documentation which informs user that they need to provide a basepath in order to use Appium CLI v < v2? 
I believe we can pass a blank/empty string to it and it still works fine.

Reply:
Just noticed that Blank String as argument values are not added to the appium command line while starting "main.js", will throw an exception if user tries to configure basepath as blank.

Below code which doesn't add value to the command line in case of blank

 ```
@Override
    protected ImmutableList<String> createArgs() {
...
Set<Map.Entry<String, String>> entries = serverArguments.entrySet();
        for (Map.Entry<String, String> entry : entries) {
...
if (!StringUtils.isBlank(value)) {
                argList.add(value);
            }
...
```

Reply:
> quick question on this - do we have any log/documentation which informs user that they need to provide a basepath in order to use Appium CLI v < v2?
> I believe we can pass a blank/empty string to it and it still works fine.

https://github.com/appium/java-client/blob/master/docs/v7-to-v8-migration-guide.md#appiumdriverlocalservice

Reply:
Fixed this in the new commit, please review.